src/main/java/com/basilisk/frontend/components/TweetDisplayComponent.java

```diff
@@ -138,10 +172,20 @@ private void commentButtonClicked() {
     @EventHandler
     private void retweetButtonClicked() {
         // Called from the template click handler
-        System.out.println("Retweet");
+        User currentUser = (User) VaadinSession.getCurrent().getAttribute(Constants.CURRENT_USER);
+        tweetPresenter.createAndSaveRetweet(tweet, currentUser);
+        UI.getCurrent().getPage().reload();
     }
 
     public interface TweetDisplayComponentModel extends TemplateModel {
         // Add setters and getters for template properties here.
     }
+
+    public static class TweetDisplayComponentByTimeStampComparator implements Comparator<TweetDisplayComponent> {
+
+        @Override
+        public int compare(TweetDisplayComponent comp1, TweetDisplayComponent comp2) {
+            return comp1.getTimeStamp().compareTo(comp2.getTimeStamp());
```

Comment:
You can reverse this by putting a negative sign after the return

Reply:
I changed the compare method to sort in reverse chronological order